File: java/src/Ui/CheckoutCommand.java
package Ui;

import Basket.Basket;
import Warehouse.Warehouse;

public class CheckoutCommand {

	private Basket basket;
	private Warehouse warehouse;

	public CheckoutCommand(Basket basket, Warehouse warehouse) {
		this.basket = basket;
		this.warehouse = warehouse;
	}

	public void run(UserInput cmd) {
		basket.checkout(warehouse);
	}

}

File: dotnet/BasketActions/CheckoutCommand.cs
namespace BasketActions {

	using Ui;

	class CheckoutCommand : UserCommand {

		private Basket.Basket basket;
		private Warehouse.Warehouse warehouse;

		internal CheckoutCommand(Basket.Basket basket, Warehouse.Warehouse warehouse) {
			this.basket = basket;
			this.warehouse = warehouse;
		}

		public void run(UserInput cmd) {
			basket.checkout(warehouse);
		}

	}
}